### DISCUSSION TITLE
When will XPS issues be fixed?

### DISCUSSION THREAD

**Discussion by wstaelens** [dotnet/wpf#6373] [General]
As you have noticed there are many XPS issues out there (most of them related to memory issues and memory leaks, fonts, STA, ...).

Over the last months, years I have seen many of them being moved to "future", "future", "future", "future" without getting proper attention or getting closed. I know that XPS and printing might not be sexy but it is needed as hell.

It is even so bad that Microsoft's own V4 print driver produces XPS jobs that are unable to read in .NET Core because code is not implemented, as @ThomasGoulet73 found out: https://github.com/dotnet/wpf/issues/6301 and https://github.com/dotnet/wpf/issues/3546 .

Why is XPS being ignored? 
Why does it take literally years before someone fixes some XPS bugs?

@danmoseley what is going on?

- https://github.com/dotnet/wpf/issues?q=xps
- https://github.com/dotnet/wpf/issues/6301 ,
- https://www.bing.com/search?q=xps+fixedpage+memory+leak+updatelayout&qs=n&form=QBRE&msbsrank=0_0__0&sp=-1&pq=xps+fixedpage+memory+leak+updatelayout&sc=0-38&sk=&cvid=AB745EF580034F38A1F9310015972FFC 
- https://stackoverflow.com/search?q=xps+memory+leak

**Comment by olowk** [comment]
We are also wandering this. In our latest project with XPS we have many problems with memory leaks and the suggested workarounds like using updatelayout don't seem to work!!

**Comment by wstaelens** [comment]
https://github.com/dotnet/runtime/issues/78629 

will anyone have a look at these top 4? Broken since the first release of .net (core).

https://github.com/dotnet/runtime/issues/51929

https://github.com/dotnet/wpf/issues/3546

https://github.com/dotnet/runtime/pull/78374

https://github.com/dotnet/wpf/issues/4000

**Reply by ThomasGoulet73** [reply]
All these issues are for dotnet/runtime. It's unfortunate that your issues don't seem to get traction there but it doesn't seem like anything can be done in the dotnet/wpf repo.

**Comment by wstaelens** [comment]
@ThomasGoulet73 guess you didn't check not everything is runtime, some are XPS/WPF/XAML. The fact is that there is a pull request that needs to be tested (code originally added by Kevin Cathcart) but this has effect on XPS/WPF project.

but really nobody seems to give a f#.

https://github.com/dotnet/wpf/issues/3546

https://github.com/dotnet/runtime/pull/78374 <-- contains the code from Kevin 

https://github.com/dotnet/wpf/issues/4000

**Reply by ThomasGoulet73** [reply]
The code related to #3546 is in dotnet/runtime so the fix will need to be in dotnet/runtime (Like dotnet/runtime#78374 attempted) with the team that owns the code (Which isn't the WPF team IIRC).

I don't have the whole context for #4000 but it seems like the original question was answered in the issue.

**Reply by wstaelens** [reply]
@ThomasGoulet73 when working with XPS most of it is in WPF but the root cause is that something from the compression (indeed runtime) is not implemented/ported fully. Everything started from WPF code base, but the issue itself is indeed in runtime, but originated from WPF code base usage. (see example: https://github.com/dotnet/wpf/issues/3546#issuecomment-695942720 )

regarding #4000 
see https://github.com/dotnet/wpf/issues/4000#issuecomment-1080367272  and https://github.com/dotnet/wpf/issues/4000#issuecomment-845712273
The problem is that everything when working with XPS documents it needs to be done in the same single STA thread. Somehow you will face the STA thread requirement which is a huge penalty.

**Comment by wstaelens** [comment]
Any updates? Most of this is in the xaml/wpf area:
https://github.com/dotnet/runtime/issues/78629

**Reply by pchaurasia14** [reply]
@wstaelens - Unfortunately, none at this time.
